zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -364,7 +371,14 @@ cdef _array_for_column_impl(object dtype,
             # of v1. However, if we look back from t6, we should see v0 for the
             # period from t0 to t1.
             end = max(non_null_ad_ixs[ix] - 1, 0)
-            if end >= asof_ix:
+            if end >= asof_ix and out_array[asof_ix, column_ix] != value:
+                # The second condition of this ``and`` is technically
```

Comment:
I'm not sure it helps to have this duplicated.  I'd probably just put a "see note above" on this.